pkg/serializer/serializer.go

```diff
@@ -60,32 +57,29 @@ var (
 func init() {
 	expvars.Set("SendEventsErrItemTooBigs", &expvarsSendEventsErrItemTooBigs)
 	expvars.Set("SendEventsErrItemTooBigsFallback", &expvarsSendEventsErrItemTooBigsFallback)
-	initExtraHeaders()
 }
 
 // initExtraHeaders initializes the global extraHeaders variables.
 // Not part of the `init` function body to ease testing
-func initExtraHeaders() {
-	jsonExtraHeaders = make(http.Header)
+func initExtraHeaders(jsonExtraHeaders, protobufExtraHeaders, jsonExtraHeadersWithCompression, protobufExtraHeadersWithCompression http.Header) {
 	jsonExtraHeaders.Set("Content-Type", jsonContentType)
 
-	jsonExtraHeadersWithCompression = make(http.Header)
 	for k := range jsonExtraHeaders {
 		jsonExtraHeadersWithCompression.Set(k, jsonExtraHeaders.Get(k))
 	}
 
-	protobufExtraHeaders = make(http.Header)
 	protobufExtraHeaders.Set("Content-Type", protobufContentType)
 	protobufExtraHeaders.Set(payloadVersionHTTPHeader, AgentPayloadVersion)
 
-	protobufExtraHeadersWithCompression = make(http.Header)
 	for k := range protobufExtraHeaders {
 		protobufExtraHeadersWithCompression.Set(k, protobufExtraHeaders.Get(k))
 	}
 
 	if compression.ContentEncoding != "" {
-		jsonExtraHeadersWithCompression.Set("Content-Encoding", compression.ContentEncoding)
-		protobufExtraHeadersWithCompression.Set("Content-Encoding", compression.ContentEncoding)
+		kind := config.Datadog.GetString("serializer_compressor_kind")
```

Comment:
we should validate that we support this compression mechanism (at startup maybe ?)